File: solutions/java/0452.java
class Solution {
  public int findMinArrowShots(int[][] points) {
    int ans = 0;
    long arrowX = (long) Integer.MIN_VALUE - 1;

    Arrays.sort(points, (a, b) -> a[1] - b[1]);

    for (int[] point : points)
      if (point[0] > arrowX) {
        ++ans;
        arrowX = point[1];
      }

    return ans;
  }
}

File: solutions/python3/0452.py
class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        ans = 0
        arrowX = float('-inf')

        for point in sorted(points, key=lambda x: x[1]):
            if point[0] > arrowX:
                ans += 1
                arrowX = point[1]

        return ans
